TestFoundation/TestDateFormatter.swift

```diff
@@ -356,4 +358,54 @@ class TestDateFormatter: XCTestCase {
         // Time zone should go back to the system one.
         XCTAssertEqual(f.timeZone, NSTimeZone.system)
     }
+
+    func test_setTimeZone() {
+        // Test two different time zones. Should ensure that if one
+        // happens to be TimeZone.current, we still get a valid test.
+        let newYork = TimeZone(identifier: "America/New_York")!
+        let losAngeles = TimeZone(identifier: "America/Los_Angeles")!
+
+        XCTAssertNotEqual(newYork, losAngeles)
+
+        // Case 1: New York
+        let f = DateFormatter()
+        f.timeZone = newYork
+        XCTAssertEqual(f.timeZone, newYork)
+
+        // Case 2: Los Angeles
+        f.timeZone = losAngeles
+        XCTAssertEqual(f.timeZone, losAngeles)
+    }
+
+    func test_ExpectedTimeZone() {
+        let gmt = TimeZone(abbreviation: DEFAULT_TIMEZONE)
+        let newYork = TimeZone(identifier: "America/New_York")!
+        let losAngeles = TimeZone(identifier: "America/Los_Angeles")!
+
+        XCTAssertNotEqual(newYork, losAngeles)
+
+        let now = Date()
+
+        let f = DateFormatter()
+        f.dateFormat = "z"
+
+        // Case 1: TimeZone.current
+        f.timeZone = TimeZone.current
+        XCTAssertEqual(f.string(from: now), f.timeZone.abbreviation())
+
+        // Case 2: New York
+        f.timeZone = newYork
+        XCTAssertEqual(f.string(from: now), f.timeZone.abbreviation())
```

Comment:
Would it make sense to us f.string(from: now), newYork.abbreviation())? Ie, if f.timeZone does not actually change the timezone, this check will succeed (with off-by-1 fix).

I understand that's not the purpose of this test case, but it is to catch that off-by-one error in the name, but I don't see a disadvantage in using the actual timezone variable instead of f.timeZone. Thoughts?

Reply:
That's fair. I generally agree that tests should avoid behavior like this. This code was mostly easier to copy/paste 3 times without errors as written, to be honest, and I forgot to clean it up after confirming the test failed without the fix and passed with it.